pkg/trace/graphbuilder.go

```diff
@@ -0,0 +1,187 @@
+package trace
+
+import (
+	"container/list"
+	"errors"
+	"fmt"
+
+	"github.com/crossplaneio/crossplane-cli/pkg/crossplane"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KubeGraphBuilder enables building graphs between kubernetes objects with the relations defined by objects.
+type KubeGraphBuilder struct {
+	client     dynamic.Interface
+	restMapper meta.RESTMapper
+	nodes      map[string]*Node
+}
+
+// NewKubeGraphBuilder returns a KubeGraphBuilder ready for building graphs.
+func NewKubeGraphBuilder(client dynamic.Interface, restMapper meta.RESTMapper) *KubeGraphBuilder {
+	return &KubeGraphBuilder{
+		client:     client,
+		restMapper: restMapper,
+		nodes:      map[string]*Node{},
+	}
+}
+
+// BuildGraph builds a graphs for the object, whose name, namespace and groupRes provided as input.
+// This method returns the root node as well as the traversed array of nodes.
+func (g *KubeGraphBuilder) BuildGraph(name, namespace, groupRes string) (root *Node, traversed []*Node, err error) {
+	queue := list.New()
+
+	traversed = make([]*Node, 0)
+
+	u := &unstructured.Unstructured{Object: map[string]interface{}{}}
+
+	gr := schema.ParseGroupResource(groupRes)
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   gr.Group,
+		Version: "",
+		Kind:    gr.Resource,
+	})
+	u.SetName(name)
+	u.SetNamespace(namespace)
+
+	root, err = g.addNodeIfNotExist(u)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = g.fetchObj(root)
+	if kerrors.IsNotFound(err) {
+		return root, nil, errors.New(
+			fmt.Sprintf("Object to trace is not found: \"%s\" \"%s\" in namespace \"%s\"", groupRes, name, namespace))
```

Comment:
Grammar nitpick: `Object to trace not found` or `Object to trace was not found` would be better. I would normally go with the `Object to trace not found` form